lib/polyfill/vttcue.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+goog.provide('shaka.polyfill.VTTCue');
+
+goog.require('shaka.log');
+goog.require('shaka.polyfill.register');
+
+
+/**
+ * @namespace shaka.polyfill.VTTCue
+ *
+ * @summary A polyfill to provide VTTCue.
+ */
+
+
+/**
+ * Install the polyfill if needed.
+ */
+shaka.polyfill.VTTCue.install = function() {
+  if (window.VTTCue) {
+    shaka.log.info('Using native VTTCue.');
+    return;
+  }
+
+  if (window.TextTrackCue &&
+      window.TextTrackCue.prototype.constructor.length == 7) {
+    shaka.log.info('Using VTTCue polyfill from 7 argument TextTrackCue.');
+    window.VTTCue = shaka.polyfill.VTTCue.from7ArgsTextTrackCue_;
+  } else {
+    shaka.log.info('VTTCue not available.');
+  }
+};
+
+
+/**
+ * Draft spec TextTrackCue with 7 constructor arguments.
+ * See {@link https://goo.gl/CSPvbz Using the TextTrack API} and examples.
```

Comment:
righ I used wrong link in the comment, this one is the one with 7 args http://www.w3schools.com/TAGs/av_met_addtexttrack.asp just like on Toshiba